reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/TimedSchedulerTest.java

```diff
@@ -0,0 +1,340 @@
+	}
+
+	@Test
+	void scheduleIncrementDirectCounterOnly() {
+		TimedScheduler test = new TimedScheduler(Schedulers.immediate(), registry, "test", Tags.empty());
+
+		test.schedule(() -> {});
+
+		assertThat(test.submittedDirect.count()).as("submittedDirect.count").isOne();
+		assertThat(test.submittedDelayed.count()).as("submittedDelayed.count").isZero();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("submittedPeriodicallyInitial.count").isZero();
+		assertThat(test.submittedPeriodicallyIteration.count()).as("submittedPeriodicallyIteration.count").isZero();
+	}
+
+	@Test
+	void scheduleDelayIncrementsDelayedCounter() throws InterruptedException {
+		TimedScheduler test = new TimedScheduler(Schedulers.single(), registry, "test", Tags.empty());
+
+		test.schedule(() -> {}, 100, TimeUnit.MILLISECONDS);
+
+		assertThat(test.submittedDirect.count()).as("submittedDirect.count").isZero();
+		assertThat(test.submittedDelayed.count()).as("submittedDelayed.count").isOne();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("submittedPeriodicallyInitial.count").isZero();
+		assertThat(test.submittedPeriodicallyIteration.count()).as("submittedPeriodicallyIteration.count").isZero();
+	}
+
+	@Test
+	void schedulePeriodicallyIncrementsPeriodicallyCountersInitialByOneAndIterationByFour() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(5);
+		TimedScheduler test = new TimedScheduler(Schedulers.single(), registry, "test", Tags.empty());
+
+		Disposable d = test.schedulePeriodically(latch::countDown,100, 100, TimeUnit.MILLISECONDS);
+
+		latch.await(10, TimeUnit.SECONDS);
+		d.dispose();
+
+		assertThat(test.submittedDirect.count()).as("submittedDirect.count").isZero();
+		assertThat(test.submittedDelayed.count()).as("submittedDelayed.count").isZero();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("submittedPeriodicallyInitial.count").isOne();
+		assertThat(test.submittedPeriodicallyIteration.count()).as("submittedPeriodicallyIteration.count").isEqualTo(4);
+	}
+
+	@Test
+	void schedulePeriodicallyIncrementsCompletedTasksByNumberOfIterations() throws InterruptedException {
+		CountDownLatch latch = new CountDownLatch(5);
+		TimedScheduler test = new TimedScheduler(Schedulers.single(), registry, "test", Tags.empty());
+
+		Disposable d = test.schedulePeriodically(latch::countDown,100, 100, TimeUnit.MILLISECONDS);
+
+		latch.await(10, TimeUnit.SECONDS);
+		d.dispose();
+
+		assertThat(test.completedTasks.count())
+			.as("completed counter tracks all iterations")
+			.isEqualTo(5)
+			.matches(l -> l == test.submittedDirect.count() + test.submittedDelayed.count()  + test.submittedPeriodicallyInitial.count()
+				+ test.submittedPeriodicallyIteration.count(), "completed tasks == sum of all timer counts");
+	}
+
+	@Test
+	void createWorkerDelegatesToAnOriginalWorker() {
+		Scheduler mockScheduler = Mockito.mock(Scheduler.class);
+		Scheduler.Worker mockWorker = Mockito.spy(Scheduler.Worker.class);
```

Comment:
Any reason you used `spy` instead of `mock` in this case?

Reply:
oops, missed that one. no good reason not to use `mock`, fixing